File: solutions/swift/triangle/2/Sources/Triangle/Triangle.swift
//Solution goes in Sources

class Triangle {
    let triangleKind = (Equilateral:"Equilateral", Isosceles:"Isosceles", Scalene:"Scalene", ErrorKind:"ErrorKind")
    var _kind: String
    init(_ a: Double, _ b: Double, _ c: Double) {
        if a <= 0 || b <= 0 || c <= 0 {
            _kind = triangleKind.ErrorKind
        } else if a + b < c || b + c < a || a + c < b {
            _kind = triangleKind.ErrorKind
        } else if a != b && b != c && c != a {
            _kind = triangleKind.Scalene
        } else if a == b && b == c {
            _kind = triangleKind.Equilateral
        } else {
            _kind = triangleKind.Isosceles
        }
    }
    
    var kind: String  {
        _kind
    }
}

File: solutions/prolog/triangle/1/triangle.pl
valid(A, B, C) :- A+B>C,A+C>B,B+C>A,A>0,B>0,C>0.

triangle(Side1, Side1, Side1, "equilateral") :- valid(Side1, Side1, Side1), !.
triangle(Side1, Side1, Side2, "isosceles") :- valid(Side1, Side1, Side2),!.
triangle(Side1, Side2, Side1, "isosceles") :- valid(Side1, Side1, Side2),!.
triangle(Side2, Side1, Side1, "isosceles") :- valid(Side1, Side1, Side2),!.
triangle(Side1, Side2, Side3, "scalene") :- valid(Side1, Side2, Side3), \+ triangle(Side1, Side2, Side3, "isosceles").